src/main/java/seedu/duke/commands/Command.java

```diff
@@ -18,6 +18,7 @@ public abstract class Command {
     protected ArrayList<Item> itemInventory;
     protected HashMap<String, ArrayList<Item>> itemNameHash;
     protected HashMap<String, Item> upcCodes;
+    protected HashMap<String, ArrayList<Item>> categoryHash;
```

Comment:
be warned: this will add a large amount of complexity and is also mostly for filterByCategory

Reply:
`filterByCategory` is still functional and standalone from the `cat [Category]` function, while they both output items from a certain category, `filterByCategory` will display each item's name, upc, quantity, price, while `cat [Category]` is more to serve as a quick lookup table with only items' name and upc code.